proxy/proxy.go

```diff
@@ -20,10 +22,11 @@ const (
 // Settings holds the values for the HTTP, HTTPS and FTP proxies as well as the
 // no_proxy value found by Detect Proxies.
 type Settings struct {
-	Http    string
-	Https   string
-	Ftp     string
-	NoProxy string
+	Http        string
+	Https       string
+	Ftp         string
+	NoProxy     string
+	AutoNoProxy string
```

Comment:
We should probably clearly document AutoNoProxy as it isn't something people who know about proxy stuff are going to immediately understand.

At least an update to the Settings doc.